timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/LockCollection.java

```diff
@@ -21,17 +21,18 @@
 
 final class LockCollection {
```

Comment:
One other general refactor I'd like to do in a separate PR which would be very quick: use design by extension, define interfaces for these objects (`LockCollection`, `MinTimestampTracker`, `MinTimestampLockCollection`)

Having interfaces makes it easier to ignore some parts of the code initially on first reads. Let me know if this sounds good